examples/pulse-mpi.py

```diff
@@ -154,21 +150,58 @@ def my_rhs(t, state):
                               boundaries=boundaries, eos=eos)
 
     def my_checkpoint(step, t, dt, state):
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              exittol=exittol, constant_cfl=constant_cfl, comm=comm)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
+        from mirgecom.simutil import check_step
+        do_status = check_step(step=step, interval=nstatus)
+        do_viz = check_step(step=step, interval=nviz)
+        do_health = check_step(step=step, interval=nhealth)
+
+        if do_status or do_viz or do_health:
+            pressure = eos.pressure(state)
+            io_fields = [
+                ("cv", state),
+                ("pressure", pressure)
+            ]
+
+        if do_status:  # This is bad, logging already completely replaces this
+            from mirgecom.io import make_status_message
+            dv = eos.dependent_vars(state)
+            # this is also bad - no option for user customization, field selection
+            status_msg = make_status_message(discr=discr, t=t, step=step, dt=dt,
+                                             cfl=current_cfl, dependent_vars=dv)
+            if rank == 0:
+                logger.info(status_msg)
```

Comment:
I'm going to echo what @inducer already mentioned once more here. Maybe it's okay to drop this whole `do_status` chunk here? If logging already replaces this, we need not duplicate efforts here. Unless you think this needs to be here for the time being?

Reply:
Nope.  I'll update these other examples.  I think `autoignition-mpi.py` was the only one that's been fully "modernized".  The one thing I will mention is that since have next to zero control over logging - when it outputs, etc, I had to add some status messaging to some cases because otherwise I was never able to see any status messages because logging doesn't choose to write them.  I hate this.

Reply:
All of the examples are updated to use logpyle now.  I did leave status in for those that make comparisons with exact solns because it is where the component errors are reported.